packages/lib/zuauth/src/server.ts

```diff
@@ -52,17 +55,27 @@ export async function authenticate(
       isEqualEdDSAPublicKey(pubKey, pcd.claim.signer)
     )
   ) {
-    throw new Error(
+    throw new ZuAuthAuthenticationError(
       "Signing key does not match any of the configured public keys"
     );
   }
 
+  if (
+    eventIds.size > 0 &&
+    pcd.claim.partialTicket.eventId &&
+    !eventIds.has(pcd.claim.partialTicket.eventId)
```

Comment:
If eventId is not revealed, you need to check whether `validEventIds` matches the configured set of `eventIds`.  Looking at the code elsewhere, it seems like this only happens if the number of `eventIds` is <= 20, which is the limit of the circuit.  That does seem to mean that if you have >20 and eventID isn't revealed, there's no secure guarantee.

Reply:
> If eventId is not revealed, you need to check whether validEventIds matches the configured set of eventIds. 

Why?

Reply:
In general, all the claims should be checked against expectations, so that a prover can't fool you by sending a proof of something other than the conditions you asked for.  That's because there's no cryptographic guarantee the prover can't change the args before generating the proof.  validEventIds is simply a part of that principal.

The "If eventId is not revealed" part of my statement is simply acknowledging that checking a single revealed event ID is stricter than validEventIds, so if eventId is revealed and checked here it's okay to skip checking validEventIds.  I'd probably go ahead and check everything, though, just to be safe and clear.

See my longer comment below for more thoughts on the interactions between the different ways of checking event ID.